src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -628,6 +648,55 @@ private void processLogsAndUpload() throws InterruptedException {
         }
     }
 
+    /**
+     * calculate the hash for the active file, which takes and digests first some lines (default is 1) in the file.
+     * If the content of file is insufficient, or incorrect setting for DEFAULT_LINES_FOR_DIGEST_NUM, the hash is null.
+     * @param activeFile
+     * @return the nullable calculated hash for active file
+     */
+    private Optional<String> calculateHashForActiveFile(File activeFile) {
+        Optional activeFileHash = Optional.ofNullable(null);
+        // String activeFileHash = null;
+        try (SeekableByteChannel chan = Files.newByteChannel(activeFile.toPath(), StandardOpenOption.READ);
+             PositionTrackingBufferedReader r =
+                new PositionTrackingBufferedReader(new InputStreamReader(Channels.newInputStream(chan),
+                        StandardCharsets.UTF_8))) {
+            r.setInitialPosition(0);
+            chan.position(0);
+            StringBuilder data = new StringBuilder(r.readLine());
+
+            // if the lines for digest is less than 1 line, skip this file and generate error.
+            if (DEFAULT_LINES_FOR_DIGEST_NUM < 1) {
+                logger.atError().cause(new InvalidDigestTargetLinesNumberException("The active log file is " +
```

Comment:
if you want to log something, you must call `.log()`.


This is _not_ an error that should be logged. No one cares that there weren't enough lines, there's absolutely nothing that anyone can do about this error.